File: 백준/Silver/1764. 듣보잡/듣보잡.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Text;

namespace AlgorithmStudy
{
    class boj1764
    {
        public static void Main(string[] args)
        {
            StreamReader sr = new StreamReader(new BufferedStream(Console.OpenStandardInput()));
            string[] input = sr.ReadLine().Split();
            int n = int.Parse(input[0]), m = int.Parse(input[1]);
            Dictionary<string, int> dic = new Dictionary<string, int>();
            StringBuilder sb = new StringBuilder();
            for (int i = 0; i < n; i++)
                dic.Add(sr.ReadLine(), 0);
            for (int i = 0; i < m; i++)
            {
                input = sr.ReadLine().Split();
                if (dic.ContainsKey(input[0]))
                    dic[input[0]]++;
            }
            sr.Close();

            List<string> names = new List<string>();
            int count = 0;
            foreach (string s in dic.Keys)
            {
                if (dic[s] > 0)
                {
                    names.Add(s);
                    count++;
                }
            }
            names.Sort();
            foreach (string s in names)
                sb.AppendLine(s);
            Console.WriteLine(count);
            Console.WriteLine(sb);
        }
    }
}

File: 백준/Silver/1764. 듣보잡/듣보잡.cc
#include <bits/stdc++.h>
#include <unordered_map>
using namespace std;

int main() {
	unordered_map<string, bool> map;
	vector<string> answer;
	int n, m;
	cin >> n >> m;
	
	for (int i = 0; i < n; i++) {
		string input;
		cin >> input;
		map[input] = false;
	}

	for (int i = 0; i < m; i++) {
		string input;
		cin >> input;
		if (map.count(input) != 0)
			answer.push_back(input);
	}

	sort(answer.begin(), answer.end());

	printf("%d\n", answer.size());
	for (string s : answer)
		printf("%s\n", s.c_str());

	return 0;
}